src/controllers/controller.bar.js

```diff
@@ -328,18 +328,13 @@ module.exports = function(Chart) {
 			var me = this;
 			var chart = me.chart;
 			var rects = me.getMeta().data;
-			var dataset = me.getDataset();
 			var ilen = rects.length;
 			var i = 0;
-			var d;
 
 			helpers.canvas.clipArea(chart.ctx, chart.chartArea);
 
 			for (; i < ilen; ++i) {
-				d = dataset.data[i];
-				if (d !== null && d !== undefined && !isNaN(d)) {
```

Comment:
Because `isNaN` returns true when you pass in an object

Also, I don't know if it's really the bar controller's responsibility to do such a check anyway. It shouldn't need to know the logic for whether the rectangle element can be drawn. That logic should be encapsulated in the rectangle element.